Final changes for acute-care files

MattWellie opened the pull request:
add new script to copy a docker image to GCP AR,
linting of a test, 
correction of logging lines & more options for subsetting,
string escaping on family-string generation script
- required when the String is handled through the analysis-runner process through to the dataproc process to retain structure

illusional commented on datasets/acute-care/families_to_samples.py:
```diff
@@ -98,6 +98,28 @@ def get_family_to_sample_map(
+    This string is used as an argument to analysis-runner, which submits it as a string argument to a dataproc process
+    By escaping we ensure the second round of arguments are received correctly
+    """
+    json_string = json.dumps(pedigree_dict, separators=(",", ":"))
+    escaped = json_string.translate(
+        str.maketrans(
+            {
+                "{": r"\{",
+                "}": r"\}",
+                "]": r"\]",
+                "[": r"\[",
+                ",": r"\,",
+                ":": r"\:",
+                '"': r'\"',
+            }
+        )
+    )
+    return escaped
```
My understanding is a double dump (ie: json.dumps the resulting string) should do the escaping. Or during the print on line 229, why not `shlex.quote` and wrap in single quotes?

Or is the problem that the analysis runner is doing a poor job of quoting arguments passed to it, because we should fix the globally if so.

MattWellie replied:
Ahh dangit, I think shlex was what I was looking for but I forgot all about it... That should do the job just fine. 

I don't think this is a typical example, as passing a string-formatted JSON object instead of a pointer to a JSON file is a nightmare. The forwarding of the string argument to a second command line invocation for the dataproc cluster shouldn't cause problems with other types of String

illusional reviewed (approved):
I've left a couple of comments that aren't critical so I've approved, but interested in your feedback about the json string quoting :)

lgruen commented on datasets/acute-care/copy_slivar_image.sh:
```diff
@@ -0,0 +1,6 @@
+set -ex
+
+gcloud auth configure-docker australia-southeast1-docker.pkg.dev
+skopeo copy docker://docker.io/brentp/slivar:latest docker://australia-southeast1-docker.pkg.dev/cpg-common/images/slivar
```
Would it be worth pinning the version here?

MattWellie replied:
it would, 

I was wondering whether it would be worth making a generic form that takes the source and target locations as arguments, so copying in new containers wouldn't require a new script to be reviewed

On the flip side, I like the idea that new additions would be reviewed...

illusional replied:
@MattWellie, definitely +1 for this, I should have picked that up in the initial review.

illusional replied:
Ah you've already done it, easy :)

lgruen commented on datasets/acute-care/families_to_samples.py:
```diff
@@ -203,7 +225,9 @@ def main(project: str, families: Tuple[str], external: bool, auth_token: str):
-    print(json.dumps(family_to_samples_dict, separators=(",", ":")))
+    json_formatted = get_escaped_json_string(pedigree_dict=family_to_samples_dict)
+    print(json_formatted)
+    # print(json.dumps(family_to_samples_dict, separators=(",", ":")))
```
Debugging leftover?

lgruen reviewed (approved):
Nit: we generally use single-quotes for strings (similar to Hail and seqr), so it might be worth changing this here as well for consistency.

Happy to leave the main review here to @illusional, he knows this part of the code base much better.

illusional commented:
There's no CI set-up for this repo, but that would be good to add as default linting to bring it back in sync. I'd be afraid of adding it to this repo due to the many other scripts already written. Might be a good candidate for the rare-disease repo?

MattWellie commented:
@illusional I've switched up the quotes and added the flake8 config, but the agreed first thing that will be done with these scripts is yanking them out into the RD repo, so I'll focus on setting up CI there

illusional commented:
@MattWellie, I think Leo's suggestion of pinning the slivar container is essential as well, pulling and pushing from a named tag (rather than latest)
